### PR TITLE
Adding publish performance tests

### PR DESCRIPTION
**Pull request by JunTaoLuo**
cc @troydai @NTaylorMullen

### CODE REVIEW

File: src/Benchmarks.Utility/Helpers/DotnetHelper.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Benchmarks.Utility.Helpers
+{
+    public class DotnetHelper
+    {
+        private readonly string _executablePath = Path.Combine("cli", "bin", "dotnet.exe");
```
Comment (JunTaoLuo) [inline src/Benchmarks.Utility/Helpers/DotnetHelper.cs]: This only works on Windows. It'll need to be updated if we plan to run on *nix

File: src/Benchmarks.Utility/Helpers/DotnetHelper.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Benchmarks.Utility.Helpers
+{
+    public class DotnetHelper
```
Comment (NTaylorMullen) [inline src/Benchmarks.Utility/Helpers/DotnetHelper.cs]: Any reason this can't all just be static?

Comment (NTaylorMullen) [inline src/Benchmarks.Utility/Helpers/DotnetHelper.cs]: I see, trying to stay consistent with dnxhelper. Alright fine as is.

File: test/Microsoft.AspNet.Tests.Performance/PublishTests.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+using Benchmarks.Framework;
+using Benchmarks.Utility.Helpers;
+using Microsoft.Extensions.PlatformAbstractions;
+using Xunit;
+
+namespace Microsoft.AspNet.Tests.Performance
+{
+    public class PublishTests : BenchmarkTestBase, IClassFixture<TestSampleManager>
+    {
+
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Tests.Performance/PublishTests.cs]: nuke empty line

File: test/Microsoft.AspNet.Tests.Performance/PublishTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            var framework = PlatformServices.Default.Runtime.RuntimeType;
+            var testName = $"{projectName}.{framework}.{nameof(DnuPublishNoSource)}";
+            var testProject = _sampleManager.PrepareSample(testName, projectName, true);
+            Assert.True(testProject != null, $"Fail to set up test project.");
+
+            var testOutput = Path.Combine(PathHelper.GetNewTempFolder(), testName);
+            Directory.CreateDirectory(testOutput);
+
+            using (Collector.StartCollection())
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Tests.Performance/PublishTests.cs]: Note: Memory collection wont work. Doesn't look like anything's hooking it up.

**Comment by NTaylorMullen** [comment]
:shipit:

File: test/Microsoft.AspNet.Tests.Performance/PublishTests.cs (hunk 4 above)
Comment (JunTaoLuo) [inline test/Microsoft.AspNet.Tests.Performance/PublishTests.cs]: I don't think we care about memory right now so it should be fine.

**Pull request merged**